### DISCUSSION TITLE
Virtualized warning: <tbody> cannot appear as a child of <div>.

### DISCUSSION THREAD

**Discussion by iakovalgaev** [table-library/react-table-library#73] [Q&A] [answered]
I take [example](https://react-table-library.com/?path=/docs/features-virtualized--large-list) from docs. When I try use `<Virtalized/>` I get 
1. "_Warning: validateDOMNesting(...): &lt;tbody&gt; cannot appear as a child of &lt;div&gt;._" Is it normal that it exists, or is it a problem worth solving?
2. Also my table is completely collapsing. But as I understood, it is necessary to explicitly set the height. Did I understand correctly that explicit height is a prerequisite Virtualized Table?
3.  `<Virtalized/>` requires specifying `rowHeight`. What to do if the row height ranges from 50px to 80px? (_for example, one row contains a short text in one, and the other contains a long text, so it is placed in three lines, and then cut off._)

A couple more ideas:
- I want to be able to add Footer to Virtualized Table (_for example, using prop `footer` in `Virtualized`_)
- It would be convenient to fix the footer with a prop `fixedFooter` in `layout` object.

**Comment by iakovalgaev** [comment]
I see that an [issue](https://github.com/table-library/react-table-library/issues/71) has already been opened for the first question.

**Reply by rwieruch** [reply]
The issue should be solved, I will check your other questions soonish.

**Comment by rwieruch** [comment, marked as answer]
1. See https://github.com/table-library/react-table-library/issues/71
2. Yes, we need an explicit height for the whole table. You should check out "Features/Fixed Header" in the documentation for finding two strategies on how to apply a height to the table. There either the container has a fixed height or the container is a flex box.
3. `rowHeight` can either be a number or a function which returns a number: 

```
type RowHeight = number | ((item: TableNode, index: number) => number);
```

There is not a great support for a Footer yet, because the company I created the library for didn't have any need for it. However, if there are more requests for it, I will tackle it :)
